dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/FdcLoggingIntegrationTest.java

```diff
@@ -0,0 +1,254 @@
+        final var contributionFileErrors = updatedIds.stream().flatMap(id ->
+                spyFactory.getContributionFileErrorOptional(contributionFileId, id).stream()).toList();
+
+        softly.assertThat(updatedIds).hasSize(3).doesNotContainNull(); // 1.
+        softly.assertThat(contributionFile.getId()).isEqualTo(contributionFileId); // 2.
+
+        softly.assertThat(logged.getContributionFileIds()).containsExactly(contributionFileId, contributionFileId, contributionFileId); // 3.
+        softly.assertThat(logged.getFdcContributionIds()).containsOnlyOnceElementsOf(updatedIds);
+
+        softly.assertThat(contributionFile.getRecordsReceived()).isIn(0, null); // 4. (either zero or NULL)
+        softly.assertThat(contributionFile.getDateReceived()).isNull();
+        softly.assertThat(contributionFile.getDateModified()).isBeforeOrEqualTo(startDate);
+        softly.assertThat(contributionFile.getUserModified()).isEqualTo("DCES");
+
+        softly.assertThat(contributionFileErrors).hasSize(3); // 5.
+        contributionFileErrors.forEach(contributionFileError -> {
+            softly.assertThat(contributionFileError.getContributionFileId()).isEqualTo(contributionFileId);
+            softly.assertThat(contributionFileError.getContributionId()).isIn(updatedIds);
+            // uncomment when FdcContribution.getMaatId() works: softly.assertThat(contributionFileError.getRepId()).isIn(repIds);
+            softly.assertThat(contributionFileError.getErrorText()).isEqualTo(ERROR_TEXT);
+            softly.assertThat(contributionFileError.getConcorContributionId()).isNull();
+            softly.assertThat(contributionFileError.getFdcContributionId()).isIn(updatedIds);
+            softly.assertThat(contributionFileError.getDateCreated()).isBetween(startDate.atStartOfDay(), endDate.atTime(LocalTime.MAX));
+        });
+    }
+
+    /**
+     * Act like a DRC acknowledging a successful or failed contribution update. This test uses MockMVC to handle CSRF
+     * dnd OAuth 2.0 login, then calls our own '/process-drc-update/contribution' endpoint like the DRC would.
+     * <p>
+     * Testing utility method.
+     */
+    private void acknowledgeFdc(final int fdcContributionId, final String errorText) throws Exception {
+        final var request = UpdateLogFdcRequest.builder().fdcId(fdcContributionId).errorText(errorText).build();
+        String json = mapper.writeValueAsString(request);
+        mockMvc.perform(post("/api/internal/v1/dces-drc-integration/process-drc-update/fdc")
+                        .with(csrf())
+                        .with(oauth2Login())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(json))
+                .andExpect(status().isOk())
+                .andExpect(content().string("The request has been processed successfully"));
+    }
+
+    private void successfulFdc(final int fdcContributionId) {
```

Comment:
Minor comment: Not sure that these two methods (successfulFdc and failedFdc) are adding much value over calling acknowledgeFdc directly instead.